accepted/2021/platform-dependencies/platform-dependencies.md

```diff
@@ -0,0 +1,433 @@
+}
+```
+
+#### Platform
+
+Each platform has its own way of identifying assets that .NET is dependent upon. For example, the Open SSL library is labeled as openssl-libs in Fedora and libssl in Debian.  For that reason, platform dependencies can't be described without identifying which platform they are for. The model handles this by describing platforms as a top-level concept to organize dependencies. Since many dependencies apply to multiple versions of a given platform, the ability to describe a versionless platform, as well as more specific versioned platforms if necessary, is possible. This is done by associating a platform with a rid. Rids are hierarchical allowing more specific platforms to be targeted which provides an ideal mechanism to be able to flexibly describe the scope of dependencies. In the same way that rids are hierarchical, so too are the platforms in this model. This allows for less repetition in the actual model content and enables platforms with more specificity to append or amend dependencies from their parent platform. Some examples of this:
+
+* A dependency is taken on a new asset that is only available in a newer version of the platform.
+* A dependency name is different in a newer version of the platform which often happens when the version of the asset is contained in its name (e.g. libicu).
+
+#### Component
+
+Within each platform can be described a set of components. These components represent logical parts of .NET that are delivered as a unit. Examples of these are shared frameworks (Microsoft.NETCore.App, Microsoft.AspNetCore.App, etc) and NuGet packages.
+
+#### Platform Dependency
+
+Each component describes the dependencies it has for the platform it is contained within. A dependency is identified by its name and type (Linux distro package, DLL).
+
+A key piece of metadata that gives context to the dependency is the "usage" field. This field is set to one of the well-known values that describes the scenario in which this dependency applies. Here are some examples of usage values:
+
+* default: Indicates that the dependency applies to a canonical app scenario (i.e. Hello World).  Note that this is specifically about a canonical app and not intended to be a description of required dependencies in the absolute sense. An example of this is libicu. While libicu is not required to run an app if it has been set to use invariant globalization, the default/canonical setting of a .NET app is that invariant globalization is set to false in which case libicu is necessary. This is why the term "default" is used rather than something like "required".
+* diagnostics: Indicates the dependency should be used in scenarios where diagnostic tools are being used such as with LTTng-UST.
+* httpsys: Indicates the dependency should be used for ASP.NET Core apps that are configured to use the HTTP.sys web server.
+* localization: Indicates the dependency should be used for localization/globalization scenarios such as with tzdata.
```

Comment:
I don't think this will scale and at the same time, it's not granular enough. I can imagine a model where each NuGet can describe its dependencies as more maintainable and scalable.

Reply:
I've made an update so that these usages are self-described within the model: https://github.com/dotnet/designs/pull/184/commits/8c52259d25a6610d8c8a1c6f228d303210575c4f.  This adds a mapping at the root of the model to define all the available dependency usages so that there can still be model validation and a way to provide a description of the value.  This will allow teams to add more usages as they need and confine the updates to just the model file.